examples/oft-solana/tasks/solana/index.ts

```diff
@@ -155,22 +155,52 @@ export const getAddressLookupTable = async (connection: Connection, umi: Umi, fr
     }
 }
 
+export enum TransactionType {
+    CreateToken = 'CreateToken',
+    CreateMultisig = 'CreateMultisig',
+    InitOft = 'InitOft',
+    SetAuthority = 'SetAuthority',
+    InitConfig = 'InitConfig',
+    SendOFT = 'SendOFT',
+}
+
+const TransactionCuEstimates: Record<TransactionType, number> = {
+    // for the sample values, they are: devnet, mainnet
+    [TransactionType.CreateToken]: 125_000, // actual sample: (59073, 123539), 55785 (more volatile as it has CPI to Metaplex)
+    [TransactionType.CreateMultisig]: 5_000, // actual sample: 3,230
+    [TransactionType.InitOft]: 70_000, // actual sample: 59207, 65198 (note: this is the only transaction that createOFTAdapter does)
+    [TransactionType.SetAuthority]: 8_000, // actual sample: 6424, 6472
+    [TransactionType.InitConfig]: 42_000, // actual sample: 33157, 40657
+    [TransactionType.SendOFT]: 230_000, // actual sample: 217,784
+}
+
 export const getComputeUnitPriceAndLimit = async (
     connection: Connection,
     ixs: Instruction[],
     wallet: KeypairSigner,
-    lookupTableAccount: AddressLookupTableAccount
+    lookupTableAccount: AddressLookupTableAccount,
+    transactionType: TransactionType
 ) => {
     const { averageFeeExcludingZeros } = await getFee(connection)
     const priorityFee = Math.round(averageFeeExcludingZeros)
     const computeUnitPrice = BigInt(priorityFee)
 
-    const computeUnits = await getSimulationComputeUnits(
-        connection,
-        ixs.map((ix) => toWeb3JsInstruction(ix)),
-        toWeb3JsPublicKey(wallet.publicKey),
-        [lookupTableAccount]
-    )
+    let computeUnits
+
+    try {
```

Comment:
thoughts on trying more than once before defaulting to estimates?

potentially using exponential backoff and retry and a couple times before defaulting

Reply:
nit: try/catch can just be condensed to

```
const computeUnits = await getSimulationComputeUnits(
  connection,
  ixs.map((ix) => toWeb3JsInstruction(ix)),
  toWeb3JsPublicKey(wallet.publicKey),
  [lookupTableAccount]
).catch(e => {
  console.error(`Error retrieving simulation compute units from RPC:`, e);
  console.log(
    `Falling back to hardcoded estimate for ${transactionType}: ${TransactionCuEstimates[transactionType]} CUs`
  );
  return TransactionCuEstimates[transactionType];
});
```

Reply:
exponential backoff before resorting to manual estimates is a good idea

Reply:
I'm retaining the try/catch syntax as it's more legible in terms of logic of happy path vs logic of unhappy path

Reply:
added the backoff usage below